### PR TITLE
Add support for instructed amounts (AmtDtls/InstdAmt/Amt)

### PR DESCRIPTION
**Pull request by manubo**

### CODE REVIEW

File: lib/camt_parser/general/transaction.rb [hunk 1]
```diff
@@ -111,19 +111,15 @@ def parse_amount
       if @xml_data.xpath('Amt').any?
         @xml_data.xpath('Amt/text()').text
       elsif @xml_data.xpath('AmtDtls').any?
-        @xml_data.xpath('AmtDtls/TxAmt/Amt/text()').text
-      else
-        nil
+        @xml_data.xpath('AmtDtls//Amt/text()').text
```
Comment (manubo) [inline lib/camt_parser/general/transaction.rb]: Matches any child Amt tag within AmtDtls

File: lib/camt_parser/general/transaction.rb [hunk 2]
```diff
@@ -111,19 +111,15 @@ def parse_amount
...
       end
     end
 
     def parse_currency
       if @xml_data.xpath('Amt').any?
         @xml_data.xpath('Amt/@Ccy').text
       elsif @xml_data.xpath('AmtDtls').any?
-        @xml_data.xpath('AmtDtls/TxAmt/Amt/@Ccy').text
-      else
```
Comment (manubo) [inline lib/camt_parser/general/transaction.rb]: Unnecessary else branch removed

**Comment by tobischo** [comment]
Thank you for the contribution

While it wasn't properly tested previously, would you mind adding a testcase for it now?

**Comment by manubo** [comment]
I have added a new fixture file that failed parsing before the changes in this PR and a test case verifying that the amount is correctly parsed.

**Comment by manubo** [comment]
Great thanks for your prompt reaction :-)

**Comment by tobischo** [comment]
Thanks for the contribution

Released as [v2.6.2](https://rubygems.org/gems/camt_parser/versions/2.6.2)

**Pull request merged**